src/components/ModalDialog/ModalDialog.js

```diff
@@ -1,57 +1,39 @@
 import React from "react";
-import "font-awesome/css/font-awesome.min.css";
-import "./ModalDialog.css";
+import Icon from "../Icon";
 import PropTypes from "prop-types";
 import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
 
-function ModalDialog(props) {
-  let header = props.header || "Alert";
-  let iconStyle = props.iconStyle || "Warning";
-  let cancelPrompt = props.cancelPrompt || "Cancel";
-  let confirmPrompt = props.confirmPrompt || "Confirm";
-
+function ModalDialog({
+  header,
+  iconStyle,
+  cancelPrompt,
+  confirmPrompt,
+  reportResult,
+  toggle,
+  showDialog,
+  prompt
+}) {
   function confirmClick() {
-    if (props.reportResult) {
-      props.reportResult(props.confirmPrompt);
-    }
+    if (reportResult) reportResult(confirmPrompt);
   }
 
   function cancelClick() {
-    if (props.reportResult) {
-      props.reportResult(props.cancelPrompt);
-    }
+    if (reportResult) reportResult(cancelPrompt);
   }
 
-  const toggle = () => {
-    if (props.toggle) {
-      props.toggle();
-    } else {
-      props.reportResult(props.cancelPrompt);
-    }
-  };
-
-  const iconStyleCheck = style => {
-    return style === iconStyle ? "icon-shown fa-stack fa-lg" : "icon-hidden";
+  const handleToggle = () => {
```

Comment:
Note that I renamed to avoid a name collision with the toggle prop. It's also a common convention to prefix event handlers with `handle`.